components/content-service/pkg/storage/gcloud.go

```diff
@@ -923,6 +923,10 @@ func (p *PresignedGCPStorage) DeleteObject(ctx context.Context, bucket string, q
 		if err == iterator.Done {
 			break
 		}
+		if err != nil {
+			log.WithField("bucket", bucket).WithField("object", attrs.Name).Error(err)
+			break
```

Comment:
Should it probably return an error instead?

Reply:
Yeah, that would be more in line with expectations.

Thinking about it, it might be nice to continue deleting, but returning the last error (if any). WDYT? @corneliusludmann

Reply:
Trying to continue deleting sounds reasonable. However, just use break will not work:

> [...] if Next returns an error other than iterator.Done, all subsequent calls will return the same error. To continue iteration, a new `ObjectIterator` must be created.

Thus, it would be great to have some kind of re-try mechanism.

Unfortunately, we don't know which error we usually get there.

Reply:
then let's return. We'd need the retry mechanism either way. Just one undeleted object is one object too many.

Reply:
> then let's return

Done.

Looking at the code again I noticed that the "ErrNotFound" at the bottom does not make any sense: it can never be reached. Maybe we should separate `itErr` and `err` in the `for` loop.
But I assume this is out-of-scope for now: We just want to learn what this error is.